### PR TITLE
Fixes #19585 - Allow csv export of nested attributes

### PR DESCRIPTION
**Pull request by tbrisker**
This is required for example to export host facet attributes.

### CODE REVIEW

File: test/unit/csv_exporter_test.rb [hunk 1]
```diff
@@ -5,6 +5,9 @@ class CsvExporterTest < ActiveSupport::TestCase
     result = CsvExporter.export(Host::Managed, [:id])
     assert_equal "Id\n", result.next
     assert_equal result.count, Host::Managed.count+1
+    assert_difference('CsvExporter.export(Host::Managed, [:id]).count') do
```
Comment (tbrisker) [inline test/unit/csv_exporter_test.rb:8]: not directly related to this but I noticed that previously the test may have had incorrect results as there were no hosts in the database.

**Comment by ohadlevy** [comment]
@tbrisker can you please provide example usage case and output? I'm just not seeing exactly how this is used yet :) - test failure seems unrelated.

**Comment by tbrisker** [comment]
@ohadlevy I need this currently for adding CSV export on content hosts, where I would want to have columns such as `content_facet.applicable_errata.count` for example.

**Comment by ShimShtein** [comment]
Nice feature! Two comments though:
1. (less critical) The auto-generated name looks a bit strange, the upper case for each method is not necessary
2. No exception handling for non-existent methods.

Examples:
```
[6] pry(main)> CsvExporter.export(Host.all, [:name, 'interfaces.first.mac']).map{ |line| line}
=> ["Name,Interfaces.First.Mac\n",
 "tommy-havier.mydomain.com,4e:11:11:11:11:11\n",
 "lucy-randles.mydomain.com,52:54:00:58:55:a7\n",
 "shari-general.mydomain.com,00:1a:4a:16:01:5d\n",
 "client.foreman.test,52:54:00:79:0e:c7\n",
 "matt-knoff.mydomain.com,00:1a:4a:16:01:52\n"]
[7] pry(main)> CsvExporter.export(Host.all, [:name, 'interfaces.mac']).map{ |line| line}
=> ["Name,Interfaces.Mac\n",
 "tommy-havier.mydomain.com,\n",
 "lucy-randles.mydomain.com,\n",
 "shari-general.mydomain.com,\n",
 "client.foreman.test,\n",
 "matt-knoff.mydomain.com,\n"]
```

**Comment by ohadlevy** [comment]
can we rename columns? for example
``` CsvExporter.export(Host.all, [:name, 'interfaces.first.mac'] ``` generates ```Name,Interfaces.Mac```

where I really want mac or mac-1 or something else.

**Comment by tbrisker** [comment]
@ohadlevy @ShimShtein - #4538 is for allowing to rename the headers, I did it in two PRs on purpose to make review easier. As for non-existing methods - I used try on purpose because you may have certain methods only available for some objects, there should be no exceptions thrown, just empty cells when there is no data.

**Comment by ShimShtein** [comment]
I would suggest underscores for default names (I know it is overridable in the other PR).

About non-existing methods: well, I wanted to protect from completely wrong paths (where method names are misspelled e.t.c.), but it can be managed through proper unit testing, so I guess it's not that problematic.

:+1: from me - tested and works.

**Comment by tbrisker** [comment]
@ohadlevy ping?

**Comment by ohadlevy** [comment]
@tbrisker I see the two PR's related in term of interface, I would prefer to finalize the discussion on #4538 first?

**Comment by tbrisker** [comment]
@ohadlevy the two PRs don't depend on each other, which is why I split them in two. Once one is merged I'll rebase the other and make whatever changes are needed.

**Comment by ohadlevy** [comment]
thanks @tbrisker

**Pull request merged**